### ISSUE TITLE
Document behavior of hybrid workflows comprehensively

### ISSUE THREAD

**Issue by bentsherman** [nextflow-io/nextflow#2984]
Inspired by this [Slack thread](https://nextflow.slack.com/archives/C02T98A23U7/p1655905727166399). The Nextflow docs mention hybrid workflows for [AWS Batch](https://www.nextflow.io/docs/latest/aws.html#hybrid-workloads) and [Google Cloud](https://www.nextflow.io/docs/latest/google.html#hybrid-execution). However it would be good to have a dedicated section on hybrid workloads:
- describe different possible combinations (local / HPC / cloud / K8s)
- show some diagrams
- caveats and limitations (especially a warning about egress cost)

This section might be easier to place after the website and docs overhaul.

**Comment by mribeirodantas** [comment]
I was discussing this with @abhi18av the other day. It's a topic I'm highly interested in, but not really knowledgeable about this (for now, I'm working on that 😄). Are you up for a call next week to do some tests/discussions @bentsherman @abhi18av? Feel free to suggest more people if you think that could benefit the discussion.

**Comment by bentsherman** [comment]
Sure, we can meet to talk about it further. I think you could experiment with some combinations to see what is possible (e.g. local to AWS, AWS to local), and that would be a good way for you to get some experience running pipelines.

**Comment by mribeirodantas** [comment]
I will soon get some training/sandboxes. That will be great to try this out!

**Comment by abhi18av** [comment]
Yup, this is a great feature which is not highlighted in the docs. Connecting with @mribeirodantas today to see if we can publish a good showcase for this.

**Comment by edmundmiller** [comment]
Stumbled on this issue.

I wish there was a simple "ssh" executor.

For example we have several GPU machines that can't be added to our cluster, but they're accessible on the network. 

The job would run just like the local except it would use `ssh username@remote_host command command_arguments` to launch in a specified directory.

Might be better as a plugin 🤔

**Comment by edmundmiller** [comment]
Wondering if I could hack something out with the [`shell` directive](https://www.nextflow.io/docs/latest/process.html#process-directive-shell)...

**Comment by bentsherman** [comment]
Might be possible with a custom executor in a plugin. You would also need to implement the commands to check the status and kill the task over ssh, in addition to submit

**Comment by edmundmiller** [comment]
Something like
```
process doRemoteThings {
    shell 'ssh', 'user@remote', '/bin/bash', '-euo', 'pipefail'

    '''
    echo "hello"
    '''
}

**Comment by bentsherman** [comment]
@christopher-hakkaart not urgent but would be nice to document this consistently across aws / azure / google at some point
